sealer/cmd/join.go

```diff
@@ -0,0 +1,96 @@
+// Copyright © 2021 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/alibaba/sealer/cert"
+
+	"github.com/alibaba/sealer/apply"
+	"github.com/alibaba/sealer/common"
+	"github.com/alibaba/sealer/logger"
+	"github.com/spf13/cobra"
+)
+
+var clusterName string
+var joinArgs *common.RunArgs
+
+func getClusterName(sealerPath string) []string {
+	files, err := ioutil.ReadDir(sealerPath)
+	if err != nil {
+		logger.Error(err)
+		os.Exit(1)
+	}
+	var clusters []string
+	for _, f := range files {
+		if f.IsDir() {
+			clusters = append(clusters, f.Name())
+		}
+	}
+	return clusters
+}
+
+var joinCmd = &cobra.Command{
+	Use:   "join",
+	Short: "join node to cluster",
+	Example: `
+join to default cluster:
+	sealer join --master x.x.x.x --node x.x.x.x
+join to cluster by cloud provider, just set the number of masters or nodes:
+	sealer join --master 2 --node 3
+`,
+	Args: cobra.ExactArgs(1),
```

Comment:
I am afraid that `sealer join --master x.x.x.x --node x.x.x.x` needs no args.

Reply:
OK, I'll fix it